app/components/tables/holidays_table/HolidaysTable.tsx

```diff
@@ -0,0 +1,168 @@
+}
+
+
+export default function NewHolidaysTable({ data }: HolidaysTableProps): JSX.Element {
+    if (data === null || data.length === 0) {
+        return (
+            <div style={{ display: "flex", justifyContent: "center" }}>
+                <p style={{ fontWeight: "bold" }}>No Data Available</p>
+            </div>
+        );
+    }
+
+    const columns = HolidaysTableHeaders;
+
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const [filtering, setFiltering] = useState<string>("");
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const [sorting, setSorting] = useState<any[]>([]);
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const [firstColumnSorting, setFirstColumnSorting] = useState<'asc' | 'desc'>('asc');
+
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const tableInstance = useReactTable({
+        columns,
+        data: data || [],
+        getCoreRowModel: getCoreRowModel(),
+        getPaginationRowModel: getPaginationRowModel(),
+        getSortedRowModel: getSortedRowModel(),
+        getFilteredRowModel: getFilteredRowModel(),
+        state: {
+            globalFilter: filtering,
+            sorting: sorting,
+        },
+        onGlobalFilterChange: setFiltering,
+        onSortingChange: setSorting,
+    });
+
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const [activePage, setActivePage] = useState(0);
+
+    const startPage = Math.max(0, activePage - 2);
+    const endPage = Math.min(tableInstance.getPageCount() - 1, startPage + 4);
+
+    const paginationButtons = [];
+    for (let i = startPage; i <= endPage; i++) {
+        paginationButtons.push(
+            <button
+                key={i}
+                className={`btn btn-outline-success ${activePage === i ? 'active' : ''}`}
+                onClick={() => {
+                    tableInstance.setPageIndex(i);
+                    setActivePage(i); // Update the active page when a button is clicked
+                }}
+            >
+                {i + 1}
+            </button>
+        );
+    }
+
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    useEffect(() => {
+        // Ensure that the sorting is always set to 'asc' when firstColumnSorting changes
+        setSorting([{ id: 'description', desc: firstColumnSorting === 'desc' }]);
+    }, [firstColumnSorting]);
+
+    return (
+        <div>
+            <div className="d-flex justify-content-end">
+                <div className="col-sm-3 my-3">
+                    <input
+                        className="form-control"
+                        type="text"
+                        value={filtering}
+                        onChange={(e) => setFiltering(e.target.value)}
+                        placeholder={"Search by Holiday, Day, Month, Year "}
+                    />
+                </div>
+            </div>
+            <hr style={{ margin: "0", border: "none", borderTop: "1px solid #ccc" }} />
```

Comment:
Let's try to avoid using the style HTML attribute and instead apply a CSS or Bootstrap class. Thanks!